adkf/ift.py

```diff
@@ -169,13 +156,22 @@ def ift_gradient_update(
     # Solve for d_adapt / d_meta
     hes_array, _ = jax.flatten_util.ravel_pytree(hes)
     cross_array, _ = jax.flatten_util.ravel_pytree(cross_derivs)
+
+    hes_matrix = hes_array.reshape(len(adapt_params_flat), len(adapt_params_flat))
+    cross_matrix = cross_array.reshape(len(adapt_params_flat), len(meta_params_flat))
+
+    if fix_singular_hessian and jnp.linalg.det(hes_matrix) == 0:
+        eig_val, eig_vec = jnp.linalg.eigh(hes_matrix)
+        new_eig_val = eig_val.at[eig_val == 0].set(1e-6)
+        hes_matrix = eig_vec @ jnp.diag(new_eig_val) @ jnp.linalg.inv(eig_vec)
```

Comment:
Isn't this a very expensive check? Maybe this is why the iteration speed is slower. I would personally not check for this. Did you find that it was necessary?